spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/GatewayAutoConfiguration.java

```diff
@@ -203,21 +204,16 @@ public ConfigurationService gatewayConfigurationService(BeanFactory beanFactory,
 	}
 
 	@Bean
-	public RouteLocator routeDefinitionRouteLocator(GatewayProperties properties,
-			List<GatewayFilterFactory> gatewayFilters,
-			List<RoutePredicateFactory> predicates,
-			RouteDefinitionLocator routeDefinitionLocator,
-			ConfigurationService configurationService) {
-		return new RouteDefinitionRouteLocator(routeDefinitionLocator, predicates,
-				gatewayFilters, properties, configurationService);
-	}
-
-	@Bean
-	@Primary
-	// TODO: property to disable composite?
-	public RouteLocator cachedCompositeRouteLocator(List<RouteLocator> routeLocators) {
```

Comment:
This is not a compatible change. We expect to be able to inject a list of `routeLocators`, this breaks that contract.

Reply:
Yeah indeed, I could inject in the method a `List<RouteLocator>` so the contract will still be the same.